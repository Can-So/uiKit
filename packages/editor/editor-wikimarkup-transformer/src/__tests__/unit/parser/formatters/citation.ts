import WikiMarkupTransformer from '../../../..';

describe('WikiMarkup => ADF Formatters - citation', () => {
  test('should detect citation mark at beginning of the line', () => {
    const wiki = '??citation?? text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should detect citation mark at end of the line', () => {
    const wiki = `This is a ??citation??
another line`;

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should detect citation mark at end of the input', () => {
    const wiki = 'This is a ??citation??';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should detect citation mark at in between of the line', () => {
    const wiki = 'This is a ??citation?? text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should detect citation mark surrounded by non alphanumeric characters', () => {
    const wiki = 'This is a (??citation??) text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should detect citation mark and ignoring the invalid closing symbol', () => {
    const wiki = 'This is a ??citation??citation?? text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should not be a citation mark surrounded by non-latin characters', () => {
    const wiki = 'This is not a 牛??citation??牛 text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should not be a citation mark if there is no space before ??', () => {
    const wiki = 'This is not a??citation?? text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should not be a citation mark if there is a space after opening ??', () => {
    const wiki = 'This is not a ?? citation?? text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should not be a citation mark if there is a space before closing ??', () => {
    const wiki = 'This is not a ??citation ?? text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });

  test('should not be a citation mark if there is not a space after closing ??', () => {
    const wiki = 'This is not a ??citation??text';

    const transformer = new WikiMarkupTransformer();
    expect(transformer.parse(wiki)).toMatchSnapshot();
  });
});
