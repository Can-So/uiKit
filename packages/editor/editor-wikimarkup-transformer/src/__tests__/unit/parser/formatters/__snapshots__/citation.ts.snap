// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WikiMarkup => ADF Formatters - citation should detect citation mark and ignoring the invalid closing symbol 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation??citation",
          "type": "text",
        },
        Object {
          "text": " text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should detect citation mark at beginning of the line 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation",
          "type": "text",
        },
        Object {
          "text": " text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should detect citation mark at end of the input 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should detect citation mark at end of the line 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation",
          "type": "text",
        },
        Object {
          "type": "hardBreak",
        },
        Object {
          "text": "another line",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should detect citation mark at in between of the line 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation",
          "type": "text",
        },
        Object {
          "text": " text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should detect citation mark surrounded by non alphanumeric characters 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a (",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "-- citation",
          "type": "text",
        },
        Object {
          "text": ") text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should not be a citation mark if there is a space after opening ?? 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is not a ?? citation?? text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should not be a citation mark if there is a space before closing ?? 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is not a ??citation ?? text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should not be a citation mark if there is no space before ?? 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is not a??citation?? text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should not be a citation mark if there is not a space after closing ?? 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is not a ??citation??text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`WikiMarkup => ADF Formatters - citation should not be a citation mark surrounded by non-latin characters 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is not a 牛??citation??牛 text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
